### PR TITLE
ci: Add debian bullseye to matrix

### PR DESCRIPTION
**Pull request by riddell-stan**
Add debian bullseye to matrix.

### CODE REVIEW

**Review by ahartikainen — APPROVED** [review]

**Comment by riddell-stan** [comment]
I'm having a hard time telling if the container is actually being used. It wasn't clear from a quick glance at the log.

**Comment by riddell-stan** [comment]
@ahartikainen any interest in taking a stab at fixing this -- i.e., getting testing working in these different containers? I think some modest additions to our GitHub Actions workflow will be required.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -28,6 +30,11 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
+      - name: Print compiler versions
+        run: |
+          gcc --version || true
+          clang --version || true
+
       - name: Install poetry
         run: pip install "poetry~=1.1.4"
```
Comment (ahartikainen) [inline .github/workflows/main.yml:39]:
```suggestion
        run: python3 -m pip install "poetry~=1.1.4"
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -18,6 +19,7 @@ jobs:
           - {runs-on: macos-10.15, python-version: 3.9}
           - {runs-on: ubuntu-20.04, container: "ubuntu:latest", python-version: 3.8}
```
Comment (ahartikainen) [inline .github/workflows/main.yml:20]:
```suggestion
```

Comment (ahartikainen) [reply .github/workflows/main.yml:20]: Apparently this image needs a bunch of configuration to get it working. (Can not install anything from pip due to some SSL problem)

**Review by ahartikainen — CHANGES_REQUESTED** [review]
This can be used to check version

    cat /etc/os-release

(It only works on Linux).

Also, there is a container setup step in the workflow log, which shows what system is used.

File: .github/workflows/main.yml (hunk 1 above)
Comment (ahartikainen) [reply .github/workflows/main.yml:39]: Those containers are missing `pip` in the main namespace.

**Comment by riddell-stan** [comment]
I think we should abandon this. It's enough to officially support Ubuntu 20.04 and higher and Debian 11 and higher. (Debian 11 is essentially the same as Ubuntu 21.04.)

I'll add a FAQ item about other distributions which have been reported as working. We can mention Fedora 34 there, for example.

**Pull request closed without merging**